ios/dtx_codec/channel.go

```diff
@@ -49,19 +49,36 @@ func (d *Channel) ReceiveMethodCall(selector string) Message {
 // MethodCall is the standard DTX style remote method invocation pattern. The ObjectiveC Selector goes as a NSKeyedArchiver.archived NSString into the
 // DTXMessage payload, and the arguments are separately NSKeyArchiver.archived and put into the Auxiliary DTXPrimitiveDictionary. It returns the response message and an error.
 func (d *Channel) MethodCall(selector string, args ...interface{}) (Message, error) {
-	payload, _ := nskeyedarchiver.ArchiveBin(selector)
 	auxiliary := NewPrimitiveDictionary()
 	for _, arg := range args {
 		auxiliary.AddNsKeyedArchivedObject(arg)
 	}
+
+	return d.methodCallWithReply(selector, auxiliary)
+}
+
+// MethodCallPrimitive is a DTX style remote method invocation pattern. The ObjectiveC Selector goes as a NSKeyedArchiver.archived NSString into the
+// DTXMessage payload, and the primitive arguments put into the Auxiliary DTXPrimitiveDictionary. It returns the response message and an error.
+func (d *Channel) MethodCallPrimitive(selector string, args ...int) (Message, error) {
```

Comment:
I think the name is a bit misleading as now the only option is to pass `int` (or actually `int32`) args. Should we rather have the method take the already constructed auxiliary like
```
func (d *Channel) MethodCallWithAuxiliary(selector string, aux PrimitiveDictionary) (Message, error) {
	return d.methodCallWithReply(selector, aux)
}
```

and in `DisableMemoryLimit`
```
	aux := dtx.NewPrimitiveDictionary()
	aux.AddInt32(int(pid))
	msg, err := p.processControlChannel.MethodCallWithAuxiliary("requestDisableMemoryLimitsForPid:", aux)
```